easybuild/tools/job/gc3pie.py

```diff
@@ -0,0 +1,195 @@
+        }
+        if env_vars:
+            extra_args['environment'] = env_vars
+        if hours:
+            extra_args['requested_walltime'] = hours*hr
+        if cores:
+            extra_args['requested_cores'] = cores
+        return Application(
+            # arguments
+            ['/bin/sh', '-c', script],
+            # no need to stage files in or out
+            inputs=[],
+            outputs=[],
+            # where should the output (STDOUT/STDERR) files be downloaded to?
+            output_dir=('%s/easybuild-jobs/%s' % (os.getcwd(), name)),
+            # capture STDOUT and STDERR
+            stdout='stdout.log',
+            join=True,
+            **extra_args
+            )
+
+    def submit(self, job, after=frozenset()):
+        """
+        Submit a job to the batch-queueing system, optionally specifying dependencies.
+
+        If second optional argument `after` is given, it must be a
+        sequence of jobs that must be successfully terminated before
+        the new job can run.
+
+        Actual submission is delayed until `commit()` is called.
+        """
+        self._jobs.add(job, after)
+
+    def commit(self):
+        """
+        End a bulk job submission.
+
+        Releases any jobs that were possibly queued since the last
+        `begin()` call.
+
+        No more job submissions should be attempted after `commit()`
+        has been called, until `begin()` is invoked again.
+        """
+        # Create an instance of `Engine` using the configuration file present
+        # in your home directory.
+        engine = create_engine()
+
+        # Add your application to the engine. This will NOT submit
+        # your application yet, but will make the engine *aware* of
+        # the application.
+        engine.add(self._jobs)
+
+        # in case you want to select a specific resource, call
+        # `Engine.select_resource(<resource_name>)`
+
+        # Periodically check the status of your application.
+        while self._jobs.execution.state != Run.State.TERMINATED:
+            # `Engine.progress()` will do the GC3Pie magic:
+            # submit new jobs, update status of submitted jobs, get
+            # results of terminating jobs etc...
+            engine.progress()
+
+            # report progress
+            stats = engine.stats(only=Application)
+            print_msg(
+                "build jobs: "
+                + str.join(", ", [
+                    ("%d %s" % (stats[state], state.lower()))
+                    for state in [
+                            'total',
+                            'SUBMITTED',
+                            'RUNNING',
+                            'ok',
+                            'failed',
+                    ]
+                    if stats[state] > 0
+                ]))
+
+            # Wait a few seconds...
+            time.sleep(30)
+
+        # final status report
+        stats = engine.stats(only=Application)
+        print_msg(
```

Comment:
in this case, `print_msg` (which will log too since a logger instance is passed), makes sense

usually, people use `eb` without `-l` that spits out the log to stdout, so we want to report that EB is still up to something rather than just hanging...